pkg/api/controller.go

```diff
@@ -3606,14 +3619,18 @@ func PaginationAfterPtr(a string) *PaginationAfter {
 	return &after
 }
 
-func writeError(w http.ResponseWriter, code int, v interface{}) {
+func writeError(w http.ResponseWriter, r *http.Request, code int, v interface{}) {
 	apiErr := Error{
 		Message: fmt.Sprint(v),
 	}
-	writeResponse(w, code, apiErr)
+	writeResponse(w, r, code, apiErr)
 }
 
-func writeResponse(w http.ResponseWriter, code int, response interface{}) {
+func writeResponse(w http.ResponseWriter, r *http.Request, code int, response interface{}) {
+	if errors.Is(r.Context().Err(), context.Canceled) {
+		w.WriteHeader(httpStatusClientClosedRequest) // Client closed request
+		return
+	}
```

Comment:
This is a good point - I can keep just this line and verify that all places goes through here.
In development the issue that not all flows goes through writeResponse - will check if it is possible and update.

Reply:
- Updated the code to use writeResponse instead of writeHeader which doesn't check the client close request.
- Kept both checks of client close request in writeResponse and in handleAPIError to exit early
- Extracted the check code as a function into a httputil package